### ISSUE TITLE
DNS over HTTPS stops working intermittently 

### ISSUE THREAD

**Issue by fatalbyte** [pi-hole/FTL#1535]
### Versions

- Pi-hole:  v5.15.5 (Latest: v5.15.5)
- AdminLTE:  v5.18.4 (Latest: v5.18.4)
- FTL:  v5.21 (Latest: v5.21)

### Platform

- OS and version:  Debian 10

- Platform:  Raspberry Pi

### Expected behavior

DNS sending requests to Cloudflare via HTTPS using a custom DNS entry of "127.0.0.1#5053"

cloudflared config file
`# Commandline args for cloudflared, using Cloudflare DNS
CLOUDFLARED_OPTS=--port 5053 --upstream https://xxxxxx.cloudflare-gateway.com/dns-query
pi@raspberrypi03:~ $ `

### Actual behavior / bug

When PiHole cannot make the request, I begin to see the following error "config error is REFUSED (EDE: network error)".  However, if I change the DNS to use Google or OpenDNS, save, then go back and set it to use my custom DNS entry again, it starts to work without issue.

### Steps to reproduce

N/A happens at random.

## Debug Token

- URL:  https://tricorder.pi-hole.net/TA1IBKPN/

### Screenshots

_If applicable, add screenshots to help explain your problem._

### Additional context

_Add any other context about the problem here._

**Comment by jfb-pihole** [comment]
> DNS sending requests to Cloudflare via HTTPS using a custom DNS entry of "127.0.0.1#5053"

This is not quite the way it works.  DNS does not go via https (an encrypted web protocol), it goes via unencrypted port 5053 in your case, after the client goes through Pi-hole on port 53.

In your case, Pi-hole is working properly and your problem is with your Cloudflared installation, not with Pi-hole.

**Comment by fatalbyte** [comment]
Thanks for the insight.  Apparently I wasn't digging in deep enough and it's been awhile since I've needed to update Cloudflared (actually just realized that my auto-update script hasn't been working - Doh!).

Version before update: cloudflared version 2020.12.0 (built 2020-12-08-2302 UTC)

Version after update: cloudflared has been updated to version 2023.3.0

If I had to guess, I bet this will fix my issue.
